src/main/java/net/dv8tion/jda/webhook/SentWebhookMessage.java

```diff
@@ -0,0 +1,220 @@
+        this.channelId = channelId;
+        this.webhookId = webhookId;
+        this.mentionedUsers = mentionedUsers;
+        this.mentionedRoles = Collections.unmodifiableList(mentionedRoles);
+        this.author = author;
+        this.type = type;
+        this.pinned = pinned;
+        this.mentionsEveryone = mentionsEveryone;
+        this.attachments = Collections.unmodifiableList(attachments);
+        this.embeds = Collections.unmodifiableList(embeds);
+    }
+
+    @Override
+    public List<User> getMentionedUsers()
+    {
+        return mentionedUsers;
+    }
+
+    @Override
+    public boolean mentionsEveryone()
+    {
+        return mentionsEveryone;
+    }
+
+    @Override
+    public boolean isEdited()
+    {
+        return false;
+    }
+
+    @Override
+    public OffsetDateTime getEditedTime()
+    {
+        return null;
+    }
+
+    @Override
+    public User getAuthor()
+    {
+        return author;
+    }
+
+
+    @Override
+    public boolean isFromType(ChannelType type)
+    {
+        return type == ChannelType.TEXT;
+    }
+
+    @Override
+    public boolean isWebhookMessage()
+    {
+        return true;
+    }
+
+    @Override
+    public List<Message.Attachment> getAttachments()
+    {
+        return attachments;
+    }
+
+    @Override
+    public List<MessageEmbed> getEmbeds()
+    {
+        return embeds;
+    }
+
+    @Override
+    public boolean isPinned()
+    {
+        return pinned;
+    }
+
+    @Override
+    public MessageType getType()
+    {
+        return type;
+    }
+
+    @Override
+    public long getIdLong()
+    {
+        return id;
+    }
+
+    /**
+     * The Snowflake id of the {@link net.dv8tion.jda.core.entities.TextChannel TextChannel} that this message was send to.
+     *
+     * @return Long containing the Id.
+     */
+    public long getChannelIdLong()
+    {
+        return channelId;
+    }
+
+    /**
+     * The Snowflake id of the webhook that send this message.
+     *
+     * @return Long containing the Id.
+     */
+    public long getWebhookIdLong()
+    {
+        return webhookId;
+    }
+
+    /**
+     * An immutable list of all mentioned {@link net.dv8tion.jda.core.entities.Role Roles} ids.
+     * <br>If none were mentioned, this list is empty.
+     *
+     * <p><b>This may include Roles from other {@link net.dv8tion.jda.core.entities.Guild Guilds}</b>
+     *
+     * @throws java.lang.UnsupportedOperationException
+     *         If this is not a Received Message from {@link net.dv8tion.jda.core.entities.MessageType#DEFAULT MessageType.DEFAULT}
+     *
+     *
+     * @return immutable list of mentioned Role ids
+     */
+    public List<Long> getMentionedRoleIds()
+    {
+        return mentionedRoles;
+    }
+
+    @Override
+    public boolean equals(Object o)
+    {
+        if (!(o instanceof SentWebhookMessage))
+            return false;
+        SentWebhookMessage oMsg = (SentWebhookMessage) o;
+        return this == oMsg || this.id == oMsg.id;
```

Comment:
This reference check is rather redundant if you do this here after the more expensive checks already happened, it would make more sense to check reference first before anything else.